### PR TITLE
ROX-22621: Introduce retry for flaky test

### PR DESCRIPTION
**Pull request by Maddosaurus**
## Description

This PR aims to counter flakes in the nongroovy test `TestStackroxNetworkFlows`.
We introduced a retry for the initial check to give the cluster more time to spin up before we run the test.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by lvalerom — APPROVED** [review]

File: tests/networkflows_test.go [hunk 1]
```diff
@@ -30,10 +30,21 @@ func TestStackroxNetworkFlows(t *testing.T) {
...
+
+	var clusters *v1.ClustersList
+	var err error
+	for retries := 20; retries > 0; retries-- {
+		clusters, err = clustersService.GetClusters(ctx, &v1.GetClustersRequest{})
+		if err == nil && len(clusters.GetClusters()) > 0 {
+			break
+		}
+		time.Sleep(5 * time.Second)
+	}
```
[the remark this answers is not in the document]
Comment (lvalerom) [reply tests/networkflows_test.go:42]: Yeah I forgot we could also use the `Eventually` function. So something like:

```golang
    var clusters *v1.ClusterList
    var err error
	require.Eventually(t, func() bool {
		clusters, err = clustersService.GetClusters(ctx, &v1.GetClustersRequest{})
		return err == 0 && len(clusters) > 0
	}, 5*time.Minute, 5*time.Second)
```

[the remark this answers is not in the document]
Comment (vikin91) [reply tests/networkflows_test.go:42]: That may be even better idea. @Maddosaurus let's maybe apply one of those two options for retrying.

File: tests/networkflows_test.go [hunk 2]
```diff
@@ -23,27 +32,24 @@ func hasEdges(graph *v1.NetworkGraph) bool {
...
-	}
+		return err == nil && len(clusters.GetClusters()) > 0
+	}, 2*time.Minute, 5*time.Second)
+
 	cancel()
 
-	require.NoError(t, err)
-	require.NotEqual(t, 0, len(clusters.GetClusters()))
+	require.NoError(s.T(), err)
+	require.NotEqual(s.T(), 0, len(clusters.GetClusters()))
```
Comment (lvalerom) [inline tests/networkflows_test.go:52]: [nit] If you're using the `suite`, you could just simply call:
```suggestion
	s.Require.NoError(err)
	s.Require.NotEqual(0, len(clusters.GetClusters()))
```

File: tests/networkflows_test.go [hunk 3]
```diff
@@ -82,7 +88,7 @@ func TestStackroxNetworkFlows(t *testing.T) {
 		}
 	}
 
-	require.NoError(t, err)
+	require.NoError(s.T(), err)
```
Comment (lvalerom) [inline tests/networkflows_test.go:91]:
```suggestion
	s.Require.NoError(err)
```

File: tests/networkflows_test.go [hunk 4]
```diff
@@ -52,7 +58,7 @@ func TestStackroxNetworkFlows(t *testing.T) {
 			break
 		}
 	}
-	require.NotNil(t, mainCluster, "cluster with name remote not found")
+	require.NotNil(s.T(), mainCluster, "cluster with name remote not found")
```
Comment (lvalerom) [inline tests/networkflows_test.go:61]: Same here:
```suggestion
	s.Require.NotNil(mainCluster, "cluster with name remote not found")
```

File: tests/networkflows_test.go [hunk 5]
```diff
@@ -64,7 +70,7 @@ func TestStackroxNetworkFlows(t *testing.T) {
 	for {
 		select {
 		case <-timeout.C:
-			t.Fatal("Failed to get the correct edges in 5 minutes")
+			s.T().Fatal("Failed to get the correct edges in 5 minutes")
```
Comment (lvalerom) [inline tests/networkflows_test.go:73]: [nit] I'm not sure if `s.FailNow` would be better here.

**Review by lvalerom — APPROVED** [review]
Added a couple of nits

File: tests/networkflows_test.go (hunk 5 above)
Comment (Maddosaurus) [reply tests/networkflows_test.go:73]: I think we're *technically* already using `FailNow`, checking the docs string on `Fatal`:
`// Fatal is equivalent to Log followed by FailNow.`
So I think we can leave it like this :smile:

**Pull request merged**